### PR TITLE
Disallow nan and inf and allow strings in csv files

### LINKED ISSUE #6298: CSV diff seems not comparing numbers with inf, nan correctly
If a user create a test with CSV diff and have divided-by-zero when creating the csv files (which will make the csv file contains inf), the test will not report error.

### PR DESCRIPTION
**Pull request by YaqiWang**
Close #6298.

### CODE REVIEW

**Comment by YaqiWang** [comment]
Anybody has comments on this? I need this in my csv tests.

File: python/TestHarness/CSVDiffer.py [hunk 1]
```diff
@@ -89,6 +89,14 @@ def diff(self):
           if abs(val2) < abs_zero:
             val2 = 0
 
+          # disallow nan in the gold file
+          if math.isnan(val1) or math.isnan(val2):
+            self.addError(fname, "The values in column \"" + key.strip() + "\" is nan")
```
Comment (andrsd) [inline python/TestHarness/CSVDiffer.py]: 1. The comment seems to be misleading. You are not allowing `NaN`s in both files. Typically the first one is the gold file (IIRC)
1. "Values is a NaN" is not correct English. You can say "Column XYZ contains NaN".

Similarly below...

**Review by andrsd — CHANGES_REQUESTED** [review]

**Comment by YaqiWang** [comment]
@andrsd comments are addressed. Any further comments?

**Review by andrsd — APPROVED** [review]

**Comment by YaqiWang** [comment]
Can this be merged?

**Pull request merged**